### DISCUSSION TITLE
Splitting a mesh on its boundaries

### DISCUSSION THREAD

**Discussion by cyubit** [pyvista/pyvista#4176] [Q&A]
Hello all, 

I have the following mesh that I am trying to split into two pieces on its boundary line. I would like to simply cut on the boundary line so that I can later delete the inside face 

<img width="434" alt="Screen Shot 2023-03-22 at 5 05 49 PM" src="https://user-images.githubusercontent.com/87676972/227038088-2a44f444-b7d2-4e01-950a-9a19d1f2b61b.png">

My current approach is to extract the mesh edges, then remove the boundary lines using remove_points(), however, this approach seems to remove too much from the edge and also does not go around the entire boundary
<img width="390" alt="Screen Shot 2023-03-22 at 5 10 44 PM" src="https://user-images.githubusercontent.com/87676972/227039034-89fae82d-9bc0-423a-9120-fa43e6e36441.png">

The goal is to remove the inside face of the mesh, as well as clean and smooth it like the following example made manually with Autodesk Meshmixer
<img width="438" alt="Screen Shot 2023-03-22 at 5 20 25 PM" src="https://user-images.githubusercontent.com/87676972/227041159-5a208d07-1260-4ed4-9024-ebd0b18196fa.png">

I've attached a ZIP with both meshes pictured here, any and all help is greatly greatly appreciated. 
[meshes.zip](https://github.com/pyvista/pyvista/files/11044881/meshes.zip)

**Comment by banesullivan** [comment]
I wonder if [vtkDecimatePro](https://vtk.org/doc/nightly/html/classvtkDecimatePro.html#details) could help with this, which we wrap in https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.decimate_pro.html#decimate-pro

**Reply by cyubit** [reply]
Thanks for your response Bane. I am currently using decimate (https://docs.pyvista.org/examples/01-filter/decimate.html) in the process of smoothing out the mesh and it's working great. However, I'm a bit of a PyVista beginner so I would love some guidance. Could you please elaborate on how I could use decimate_pro in splitting the mesh?
